src/Functions/ztest.cpp

```diff
@@ -0,0 +1,239 @@
+            }
+
+            return getReturnType();
+        }
+
+
+        ColumnPtr executeImpl(const ColumnsWithTypeAndName & const_arguments, const DataTypePtr &, size_t input_rows_count) const override
+        {
+            auto arguments = const_arguments;
+            /// Only last argument have to be constant
+            for (size_t i = 0; i < 5; ++i)
+                arguments[i].column = arguments[i].column->convertToFullColumnIfConst();
+
+            static const auto uint64_data_type = std::make_shared<DataTypeNumber<UInt64>>();
+
+            auto column_successes_x = castColumnAccurate(arguments[0], uint64_data_type);
+            const auto & data_successes_x = checkAndGetColumn<ColumnVector<UInt64>>(column_successes_x.get())->getData();
+
+            auto column_successes_y = castColumnAccurate(arguments[1], uint64_data_type);
+            const auto & data_successes_y = checkAndGetColumn<ColumnVector<UInt64>>(column_successes_y.get())->getData();
+
+            auto column_trials_x = castColumnAccurate(arguments[2], uint64_data_type);
+            const auto & data_trials_x = checkAndGetColumn<ColumnVector<UInt64>>(column_trials_x.get())->getData();
+
+            auto column_trials_y = castColumnAccurate(arguments[3], uint64_data_type);
+            const auto & data_trials_y = checkAndGetColumn<ColumnVector<UInt64>>(column_trials_y.get())->getData();
+
+            static const auto float64_data_type = std::make_shared<DataTypeNumber<Float64>>();
+
+            auto column_confidence_level = castColumnAccurate(arguments[4], float64_data_type);
+            const auto & data_confidence_level = checkAndGetColumn<ColumnVector<Float64>>(column_confidence_level.get())->getData();
+
+            if (!isColumnConst(*arguments[5].column.get()))
+                throw Exception{ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                                "The sixth argument {} of function {} should be a constant string", arguments[5].type->getName(), getName()};
+
+            String usevar = checkAndGetColumnConst<ColumnString>(arguments[5].column.get())->getValue<String>();
+
+            auto res_z_statistic = ColumnFloat64::create();
+            auto & data_z_statistic = res_z_statistic->getData();
+            data_z_statistic.reserve(input_rows_count);
+
+            auto res_p_value = ColumnFloat64::create();
+            auto & data_p_value = res_p_value->getData();
+            data_p_value.reserve(input_rows_count);
+
+            auto res_ci_lower = ColumnFloat64::create();
+            auto & data_ci_lower = res_ci_lower->getData();
+            data_ci_lower.reserve(input_rows_count);
+
+            auto res_ci_upper = ColumnFloat64::create();
+            auto & data_ci_upper = res_ci_upper->getData();
+            data_ci_upper.reserve(input_rows_count);
+
+            auto insert_values = [&data_z_statistic, &data_p_value, &data_ci_lower, &data_ci_upper](Float64 z_stat, Float64 p_value, Float64 lower, Float64 upper)
```

Comment:
Maybe better to rename into `insert_values_into_result` because when calling `insert_values` it is hard to understand that this will modify result.